src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -231,6 +233,167 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNew
+    }
+    if (source.HasAntialiasingMode() || source.AntialiasingModeOverrideSource() != nullptr)
+    {
+        duplicated->AntialiasingMode(source.AntialiasingMode());
+    }
+    if (source.HasRetroTerminalEffect() || source.RetroTerminalEffectOverrideSource() != nullptr)
+    {
+        duplicated->RetroTerminalEffect(source.RetroTerminalEffect());
+    }
+    if (source.HasForceFullRepaintRendering() || source.ForceFullRepaintRenderingOverrideSource() != nullptr)
+    {
+        duplicated->ForceFullRepaintRendering(source.ForceFullRepaintRendering());
+    }
+    if (source.HasSoftwareRendering() || source.SoftwareRenderingOverrideSource() != nullptr)
+    {
+        duplicated->SoftwareRendering(source.SoftwareRendering());
+    }
+    if (source.HasColorSchemeName() || source.ColorSchemeNameOverrideSource() != nullptr)
+    {
+        duplicated->ColorSchemeName(source.ColorSchemeName());
+    }
+    if (source.HasForeground() || source.ForegroundOverrideSource() != nullptr)
+    {
+        duplicated->Foreground(source.Foreground());
+    }
+    if (source.HasBackground() || source.BackgroundOverrideSource() != nullptr)
+    {
+        duplicated->Background(source.Background());
+    }
+    if (source.HasSelectionBackground() || source.SelectionBackgroundOverrideSource() != nullptr)
+    {
+        duplicated->SelectionBackground(source.SelectionBackground());
+    }
+    if (source.HasCursorColor() || source.CursorColorOverrideSource() != nullptr)
+    {
+        duplicated->CursorColor(source.CursorColor());
+    }
+    if (source.HasHistorySize() || source.HistorySizeOverrideSource() != nullptr)
+    {
+        duplicated->HistorySize(source.HistorySize());
+    }
+    if (source.HasSnapOnInput() || source.SnapOnInputOverrideSource() != nullptr)
+    {
+        duplicated->SnapOnInput(source.SnapOnInput());
+    }
+    if (source.HasAltGrAliasing() || source.AltGrAliasingOverrideSource() != nullptr)
+    {
+        duplicated->AltGrAliasing(source.AltGrAliasing());
+    }
+    if (source.HasCursorShape() || source.CursorShapeOverrideSource() != nullptr)
+    {
+        duplicated->CursorShape(source.CursorShape());
+    }
+    if (source.HasCursorHeight() || source.CursorHeightOverrideSource() != nullptr)
+    {
+        duplicated->CursorHeight(source.CursorHeight());
+    }
+    if (source.HasBellStyle() || source.BellStyleOverrideSource() != nullptr)
+    {
+        duplicated->BellStyle(source.BellStyle());
+    }
+    if (source.HasPixelShaderPath() || source.PixelShaderPathOverrideSource() != nullptr)
+    {
+        duplicated->PixelShaderPath(source.PixelShaderPath());
+    }
+    if (source.HasBackgroundImageAlignment() || source.BackgroundImageAlignmentOverrideSource() != nullptr)
+    {
+        duplicated->BackgroundImageAlignment(source.BackgroundImageAlignment());
+    }
+    if (source.HasConnectionType())
+    {
+        duplicated->ConnectionType(source.ConnectionType());
+    }

```

Comment:
nit: it would be nice if we had all of this copying over in `Profile` (just makes more sense). That way we can consolidate all of the stuff that needs updating when we add a new profile setting in one file.

I would also add a big ol' comment to that new function explaining how this differs from `Profile::CopySettings` and when to use one over the other.

Reply:
I wonder if we could be clever here too to avoid another place to update when we add a new setting. Maybe we could update `Profile::CopySettings(Profile source)` to be something like this...

```c++
// - source: the profile we're copying settings from
// - checkOverrideSource: if true, copy the setting over even if it's inherited. Otherwise, only copy explicitly set values
Profile::CopySettings(com_ptr<Profile> source, bool checkOverrideSource = false)
{
	auto profile{ winrt::make_self<Profile>() };
	if (source->HasFontWeight() || (checkOverrideSource && source->FontWeightOverrideSource() != nullptr))
    {
        profile->FontWeight(source.FontWeight());
    }
	// ...
	// repeat this for every setting
	// ...
}
```

It looks nasty, but `checkOverrideSource=false` makes this act like the old `CopySettings()`, whereas `true` makes this act like the new `DuplicateProfile()`.

Be sure to set `_Origin=Custom` though.

Reply:
Also, stop introducing boolean optional parameters please. Maybe even be careful introducing boolean parameters at all? It's crazy to have functions that you call with `Thing(false, true, false, false)` and `Thing(false)` and `Thing(false, false, false, false, false, false)`